s3torchconnector/src/s3torchconnector/_user_agent.py

```diff
@@ -0,0 +1,24 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+from typing import List
+
+from ._version import __version__
+
+# https://www.rfc-editor.org/rfc/rfc9110#name-user-agent
+
+
+class UserAgent:
+    def __init__(self, comments: List[str] = None):
+        self._user_agent_prefix = f"{__package__}/{__version__}"
+        self._comments = comments or []
+        self._comments_separator = "; "
+
+    def add_comment(self, comment):
+        self._comments.append(comment)
+
+    @property
+    def prefix(self):
+        comments_str = f"{self._comments_separator.join(self._comments)}" or ""
```

Comment:
No need for the or block here, or the f-string. I'm also not convinced we need the `_comments_separator` variable at all